webrtc/RTCPeerConnectionIceEvent-constructor.html

```diff
@@ -1,34 +1,148 @@
-<!DOCTYPE html>
+<!doctype html>
 <meta charset="utf-8">
-<title>RTCPeerConnectionIceEvent constructor</title>
+<!--
+4.8.2 RTCPeerConnectionIceEvent
+
+  The icecandidate event of the RTCPeerConnection uses the RTCPeerConnectionIceEvent interface.
+
+-->
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <script>
-test(function() {
-  assert_equals(RTCPeerConnectionIceEvent.length, 1);
-  var e = new RTCPeerConnectionIceEvent('type');
-  assert_equals(e.candidate, null);
-  assert_false(e.bubbles);
-  assert_false(e.cancelable);
-}, 'RTCPeerConnectionIceEvent constructor with no candidate provided.');
-
-test(function() {
-  var e = new RTCPeerConnectionIceEvent('type', { candidate: null });
-  assert_equals(e.candidate, null);
-}, 'RTCPeerConnectionIceEvent constructor with candidate passed as null.');
-
-test(function() {
-  var e = new RTCPeerConnectionIceEvent('type', { candidate: undefined });
-  assert_equals(e.candidate, null);
-}, 'RTCPeerConnectionIceEvent constructor with candidate passed as undefined.');
-
-test(function() {
-  var c = new RTCIceCandidate({ candidate: 'candidate', sdpMid: 'sdpMid', sdpMLineIndex: 1 });
-  var e = new RTCPeerConnectionIceEvent('type', { candidate: c, url: 'url', bubbles: true, cancelable: true});
-  assert_equals(e.type, 'type');
-  assert_equals(e.candidate, c);
-  // assert_equals(e.url, 'url');
-  assert_true(e.bubbles);
-  assert_true(e.cancelable);
-}, 'RTCPeerConnectionIceEvent constructor with full arguments.');
+/*
+RTCPeerConnectionIceEvent
+
+[Constructor(DOMString type, optional RTCPeerConnectionIceEventInit eventInitDict)]
+
+interface RTCPeerConnectionIceEvent : Event {
+    readonly attribute RTCIceCandidate? candidate;
+    readonly attribute DOMString?       url;
+};
+ */
+test(() => {
+  assert_throws(new TypeError(), () => {
+    new RTCPeerConnectionIceEvent();
+  });
+}, "RTCPeerConnectionIceEvent with no arguments throws TypeError");
+
+test(() => {
+  const event = new RTCPeerConnectionIceEvent("type");
+  /*
+  candidate of type RTCIceCandidate, readonly, nullable
+  url of type DOMString, readonly, nullable
+  */
+  assert_equals(event.candidate, null);
+  assert_equals(event.url, null);
+
+  /*
+  Firing an RTCPeerConnectionIceEvent event named e with an RTCIceCandidate candidate means that an event with the name e, which does not bubble (except where otherwise stated) and is not cancelable (except where otherwise stated),
```

Comment:
The sentence should be split into multiple comment lines for easier readability.